frontera/contrib/backends/cassandra/components.py

```diff
@@ -0,0 +1,359 @@
+    def schedule(self, batch):
+        query = self.session.prepare("INSERT INTO queue (crawl, fingerprint, score, partition_id, host_crc32, url, "
+                                     "created_at, meta, depth, headers, method, cookies) "
+                                     "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)")
+        cql_items = []
+        for fprint, score, request, schedule in batch:
+            if schedule:
+                _, hostname, _, _, _, _ = parse_domain_from_url_fast(request.url)
+                if not hostname:
+                    self.logger.error("Can't get hostname for URL %s, fingerprint %s" % (request.url, fprint))
+                    partition_id = self.partitions[0]
+                    host_crc32 = 0
+                else:
+                    partition_id = self.partitioner.partition(hostname, self.partitions)
+                    host_crc32 = get_crc32(hostname)
+                created_at = time()*1E+6
+
+                if "domain" not in request.meta:
+                    request.meta["domain"] = {}
+                if "origin_is_frontier" not in request.meta:
+                    request.meta["origin_is_frontier"] = ''
+                if "scrapy_callback" not in request.meta:
+                    request.meta["scrapy_callback"] = None
+                if "scrapy_errback" not in request.meta:
+                    request.meta["scrapy_errback"] = None
+                if "scrapy_meta" not in request.meta:
+                    request.meta["scrapy_meta"] = {}
+                if "score" not in request.meta:
+                    request.meta["score"] = 0
+                if "jid" not in request.meta:
+                    request.meta["jid"] = 0
+
+                meta = Meta(domain=request.meta['domain'], fingerprint=fprint,
+                            origin_is_frontier=request.meta['origin_is_frontier'],
+                            scrapy_callback=request.meta['scrapy_callback'],
+                            scrapy_errback=request.meta['scrapy_errback'], scrapy_meta=request.meta['scrapy_meta'])
+
+                cql_i = (self.crawl_id, fprint, score, partition_id, host_crc32, request.url, created_at, meta, 0,
+                         request.headers, request.method, request.cookies)
+                cql_items.append(cql_i)
+
+                request.meta['state'] = States.QUEUED
+
+        execute_concurrent_with_args(self.session, query, cql_items, concurrency=400)
+        self.cass_count({"queued_urls": len(cql_items)})
+
+    @retry_and_rollback
+    def count(self):
+        count = self.queue_model.objects.filter(crawl=self.crawl_id).count()
+        return count
+
+    def cass_count(self, counts):
+        for row, count in counts.iteritems():
+            count_page = self.session.prepare("UPDATE crawlstats SET " + row + " = " + row + " + ? WHERE crawl= ?")
+            self.session.execute_async(count_page, (count, self.crawl_id))
+
+
+class BroadCrawlingQueue(Queue):
```

Comment:
Oh, that sounds hard for me - i will take a look at this when i've more time, ok?